app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyRollouts.controller.js

```diff
@@ -10,68 +12,55 @@ module.exports = angular
-          return promotion.scopes.from.appId.includes(vm.applicationFilter);
-        });
-      } else {
-        vm.filteredPromotions = vm.promotions;
-      }
-    };
+    vm.application = undefined;
 
-    vm.continue = function(promotionId) {
-      fastPropertyWriter.continuePromotion(promotionId).then(vm.loadPromotions);
-    };
+    vm.fetchApplication = () => {
+      applicationReader.getApplication('spinnakerfp')
+        .then((application) => {
+          vm.application = application;
+          return vm.application;
+        })
+        .then((application) => {
+          application.executions.activate();
+          return application;
+        })
+        .then((application) => {
+          application.executions.onRefresh($scope, normalizeExecutionNames, dataInitializationFailure);
 
-    vm.stop = function(promotionId) {
-      $log.warn('Stop with: ' + promotionId);
+        });
     };
 
-    vm.extractScopeFromHistoryMessage = FastPropertyScopeService.extractScopeFromHistoryMessage;
+    vm.fetchApplication();
 
-    vm.getLastMessage = function(promotion) {
-      return FastPropertyScopeService.extractScopeFromHistoryMessage(_.chain(promotion.history).last().value().message);
+    let dataInitializationFailure = () => {
+      this.viewState.loading = false;
+      this.viewState.initializationError = true;
     };
 
-
-    vm.updateStateFilter = function(state) {
-      if(state) {
-        vm.filteredPromotions = vm.promotions.filter(function(promotion) {
-          return promotion.state === state;
-        });
-      } else {
-        vm.filteredPromotions = vm.promotions;
+    function normalizeExecutionNames() {
+      if (vm.application.executions.loadFailure) {
+        dataInitializationFailure();
       }
-    };
-
-    vm.loadPromotions = function loadPromotions() {
-      fastPropertyReader.loadPromotions()
-        .then(function(promotionList) {
-          vm.promotions = vm.filteredPromotions = promotionList;
-          vm.filter();
-          return vm.promotions;
-        })
-        .then(fastPropertyTransformer.sortRunningPromotionsFirst)
-        .then(function(sortedPromotions) {
-          vm.promotions = sortedPromotions;
-          return vm.promotions;
-        })
-        .then(function() {
-          return vm.updateStateFilter(vm.promotionStateFilter);
-        }).catch(function(error) {
-          $log.warn(error);
-        });
-    };
-
-    vm.loadPromotions();
+      let executions = vm.application.executions.data || [];
+      var configurations = vm.application.pipelineConfigs.data || [];
+      executions.forEach(function(execution) {
+        if (execution.pipelineConfigId) {
+          var configMatches = configurations.filter(function(configuration) {
+            return configuration.id === execution.pipelineConfigId;
+          });
+          if (configMatches.length) {
+            execution.name = configMatches[0].name;
+          }
+        }
+      });
```

Comment:
This method seems to be copied from https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/delivery/executions/executions.controller.js#L88-L104 - probably worth moving it into a service.